core/src/main/java/uk/ac/bbsrc/tgac/miso/core/util/EraSubmissionPreparation.java

```diff
@@ -0,0 +1,496 @@
+    }
+
+    public abstract void generateDocument(ZipOutputStream output)
+        throws ParserConfigurationException, IOException, TransformerConfigurationException, TransformerException,
+        TransformerFactoryConfigurationError;
+
+    public abstract boolean isEmpty();
+
+    public abstract String name();
+  }
+
+  private final class DilutionXmlSubfile extends XmlSubmissionFromSet<Pair<PoolableElementView, Partition>> {
+
+    @Override
+    protected Stream<Pair<PoolableElementView, Partition>> items() {
+      return submission.getExperiments().stream().flatMap(experiment -> experiment.getRunPartitions().stream()).flatMap(
+          rp -> rp.getPartition().getPool().getPoolableElementViews().stream().map(poolable -> new Pair<>(poolable, rp.getPartition())));
+    }
+
+    @Override
+    public String name() {
+      return "run";
+    }
+
+    @Override
+    protected void populate(Element xml, Pair<PoolableElementView, Partition> entry) {
+      Run r = entry.getValue().getSequencerPartitionContainer().getLastRun();
+
+      xml.setAttribute("alias",
+          "L00" + entry.getValue().getPartitionNumber() + ":" + entry.getKey().getDilutionName() + ":" + r.getAlias());
+      xml.setAttribute("run_center", centreName);
+      if (r.getHealth() == HealthType.Completed) {
+        xml.setAttribute("run_date", DF_TIMESTAMP.format(r.getCompletionDate()));
+      }
+      xml.setAttribute("center_name", centreName);
+
+      for (Experiment e : submission.getExperiments()) {
+        Element experimentRef = xml.getOwnerDocument().createElementNS(null, "EXPERIMENT_REF");
+        experimentRef.setAttribute("refname", e.getAlias());
+        experimentRef.setAttribute("refcenter", centreName);
+        xml.appendChild(experimentRef);
+      }
+
+      Element dataBlock = xml.getOwnerDocument().createElementNS(null, "DATA_BLOCK");
+      dataBlock.setAttribute("sector", Integer.toString(entry.getValue().getPartitionNumber()));
+      if (entry.getValue().getPool().getPoolableElementViews().size() > 1) {
+        // multiplexed
+        dataBlock.setAttribute("member_name", entry.getKey().getDilutionName());
+      }
+
+    }
+
+    @Override
+    protected String xmlCollectionName() {
+      return "RUN_SET";
+    }
+
+    @Override
+    protected String xmlName() {
+      return "RUN";
+    }
+  }
+
+  private final class ExperimentXmlSubfile extends XmlSubmissionFromSet<Experiment> {
+
+    @Override
+    protected Stream<Experiment> items() {
+      return submission.getExperiments().stream();
+    }
+
+    @Override
+    public String name() {
+      return "experiment";
+    }
+
+    @Override
+    protected void populate(Element xml, Experiment experiment) {
+      xml.setAttribute("alias", experiment.getAlias());
+      xml.setAttribute("center_name", centreName);
+
+      Element xmlTitle = xml.getOwnerDocument().createElementNS(null, "TITLE");
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "TITLE" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)